chia/full_node/mempool_check_conditions.py

```diff
@@ -155,6 +155,10 @@ def parse_aggsig(args: SExp) -> List[bytes]:
         raise ValidationError(Err.INVALID_CONDITION)
     if len(message) > 1024:
         raise ValidationError(Err.INVALID_CONDITION)
+    # agg sig conditions only take 2 parameters
+    args = args.rest()
+    if args.pair is not None:
```

Comment:
```suggestion
    if args.atom != b"":
```
Or is this more strict due to the original using `.as_atom_list()`?

Reply:
right, the main reason is to be compatible with `as_atom_list()` I just added a comment. Another (potential) issue us that if `atom` is `None`, believe you'll get an exception in that comparison, which might lead to getting a different error code than `INVALID_CONDITION`. Our test infrastructure makes it tricky to test this in a full block-chain simulation though. I can add a unit test for it